python/ray/data/_internal/util.py

```diff
@@ -745,7 +745,7 @@ def find_partition_index(
             col_vals[null_mask] = NULL_SENTINEL
 
         prevleft = left
```

Comment:
can we write some tests for this method in particular? like test_util.py

Reply:
also ... is this even implemented properly in the first place? `desired` is from `boundary` which is in Key space, not a column index (though it is used as a column index...

Reply:
> though it is used as a column index...

Not sure I follow. If a user passed `boundaries=[-10, 0, 2, 5]` then `desired` will take on the values `(-10,)`, `(0,)`, `(2,)`, and `(5,)`. Isn't this correct?

Reply:
I'm primarily thinking about the code above:

```
for i in range(len(desired)): # L730 - i is 0 to the number of boundaries provided
...
    col_name = columns[i]  # L733 - why are we using i to index columns

Reply:
if you trace back `columns`, it comes from `sort_key.get_columns()` which returns the columns passed to `ds.sort`—i.e., `columns` is the list of columns to be sorted. 

This is also why indexing `descending[i]` is okay. We have `len(columns) == len(boundaries) == len(descending)`

Reply:
why do we assume `We have len(columns) == len(boundaries)`? Can't boundaries be arbitrary?

Reply:
yes that is definitely correct and I don't know why I said that

Reply:
resolved offline: not an issue

TODO for these two functions in `_util`: 
- update types
- add documentation
- add unit tests (especially to verify that the `partition[i]` in `return` is correct